### DISCUSSION TITLE
add exclude option to autoDiscoverFilter

### DISCUSSION THREAD

**Discussion by thetoolsmith** [renovatebot/renovate#12526] [Suggest an Idea]
### What would you like Renovate to be able to do?

We use autoDiscoverFilter to scan all repos in a particular project in source (azure devops Git). This is scheduled to run everyday with renovatebot.
We should have the option to exclude repositories in this high level filter scan.

### If you have any ideas on how this should be implemented, please tell us here.

add a config option autoDiscoverFilterExclude to allow a list of repositories to exclude, or a filter to allow patterns for excluding some repos.

### Is this a feature you are interested in implementing yourself?

No

**Comment by rarkins** [comment]
`autodiscoverFilter` uses minimatch, which supports negations: https://github.com/isaacs/minimatch

**Comment by viceice** [comment]
Your probably interested in:
- #8763

There was a PR to implement that, but it had a bug and was reverted.

**Reply by thetoolsmith** [reply]
The array would work as well. I proposed the separate config property as it's a bit more standard approach in the software tools I'm used to working with or developing.
I'm not familiar with typescript, so I'm also not that familiar with minimatch. @rarkins thanks for the link. Renovate is the only piece of code I'm exposed to using these technologies.
